### ISSUE TITLE
useing gpu to train it emerge many nan and zero

### ISSUE THREAD

**Issue by jssyzsfzy** [ultralytics/yolov5#4538]
## ❔Question
when use cpu to train models,it can run perfectly,but when use gpu,the process of training is wrong
## Additional context
when use gpu, datas(box, obj, cls) are nan, while the Labels, P, R, mAP are all zero.
cpu it can run perfectly

**Comment by jssyzsfzy** [comment]
my gpu is RTX 1650, CUDA 11.4, cudnnv8.2.2.26
python3.9, torch1.9.0
GPU:
![QQ图片20210825190444](https://user-images.githubusercontent.com/89206311/130779975-5d8d5614-0470-4f74-bf50-8d9e72528aca.jpg)
CPU:
![QQ图片20210825190449](https://user-images.githubusercontent.com/89206311/130780104-a6bc4aa4-26f6-4790-b836-761c74644f73.png)

**Comment by MangoloD** [comment]
do you solve it

**Comment by xiegongsheldon** [comment]
do you solve it?

**Comment by glenn-jocher** [comment]
@jssyzsfzy @MangoloD @xiegongsheldon some environments suffer from CUDA/GPU errors, particularly on windows and/or conda. Before you train on GPU you should run a simple test to verify that GPU inference works correctly:

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # or yolov5m, yolov5l, yolov5x, custom

# Images
img = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, PIL, OpenCV, numpy, list

# Inference
results = model(img)

# Results (should show 2 persons, 1 tie)
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
```

Please ensure you meet all dependency requirements if you are attempting to run YOLOv5 locally.  If in doubt, create a new virtual Python 3.8 environment, clone the latest repo (**code changes daily**), and `pip install -r requirements.txt` again. We also highly recommend using one of our verified environments below. 


## Requirements

[**Python>=3.6.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.7**](https://pytorch.org/get-started/locally/). To get started:
```bash
$ git clone https://github.com/ultralytics/yolov5
$ cd yolov5
$ pip install -r requirements.txt
```

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), validation ([val.py](https://github.com/ultralytics/yolov5/blob/master/val.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.
